src/Cli/dotnet/Commands/Test/TestingPlatformCommand.cs

```diff
@@ -88,6 +89,9 @@ private void PrepareEnvironment(ParseResult parseResult, out TestOptions testOpt
         testOptions = GetTestOptions(parseResult, filterModeEnabled, isHelp: ContainsHelpOption(arguments));
 
         _isDiscovery = ContainsListTestsOption(arguments);
+
+        // This is ugly, and we need to replace it by passing out some info from testing platform to inform us that some process level retry plugin is active.
+        _isRetry = arguments.Contains("--retry-failed-tests");
```

Comment:
This can come through a response file as well ;)
I see no guarantee that this is *our* retry though. A custom framework could add this command-line option, and user may not be using our Retry extension at all. Also if we ended up fixing https://github.com/microsoft/testfx/issues/5239, that will be another case where this would break.

I personally don't think this is the right way.

Reply:
I also don't think this is the right way, but I would not block shipping because of it.

issue to refine: https://github.com/microsoft/testfx/issues/5361